crypto/evp/kdf_lib.c

```diff
@@ -70,66 +104,51 @@ void EVP_KDF_reset(EVP_KDF_CTX *ctx)
         return;
 
     if (ctx->meth->reset != NULL)
-        ctx->meth->reset(ctx->impl);
+        ctx->meth->reset(ctx->data);
 }
 
-int EVP_KDF_ctrl(EVP_KDF_CTX *ctx, int cmd, ...)
+size_t EVP_KDF_size(EVP_KDF_CTX *ctx)
 {
-    int ret;
-    va_list args;
-
-    va_start(args, cmd);
-    ret = EVP_KDF_vctrl(ctx, cmd, args);
-    va_end(args);
+    if (ctx == NULL)
+        return 0;
 
-    if (ret == -2)
-        EVPerr(EVP_F_EVP_KDF_CTRL, EVP_R_COMMAND_NOT_SUPPORTED);
+    if (ctx->meth->size == NULL)
+        return SIZE_MAX;
 
-    return ret;
+    return ctx->meth->size(ctx->data);
```

Comment:
Shouldn't this be a param?

Reply:
Yeah, it probably should be a context param.

Reply:
I've switched it over to being a param.